openmc/material.py

```diff
@@ -795,16 +795,30 @@ def get_elements(self):
 
         return sorted({re.split(r'(\d+)', i)[0] for i in self.get_nuclides()})
 
-    def get_nuclides(self):
-        """Returns all nuclides in the material
+    def get_nuclides(self, element: Optional[str] = None):
+        """Returns a list of all nuclides in the material, if the element
+        argument is specified then just nuclides of that element are returned.
+
+        Parameters
+        ----------
+        element : str
+            Specifies the element to match when searching through the nuclides
 
         Returns
         -------
         nuclides : list of str
             List of nuclide names
-
         """
-        return [x.name for x in self._nuclides]
+
+        if element:
+            matching_nuclides = []
+            for nuclide in self._nuclides:
+                if re.split(r'(\d+)', nuclide.name)[0] == element:
+                    matching_nuclides.append(nuclide.name)
+            return list(set(matching_nuclides))
+
+        else:
+            return list(set([x.name for x in self._nuclides]))
```

Comment:
Unfortunately `set(...)` doesn't preserve insertion order. You could either 1) use a dict, which guarantees preservation of insertion order as of Py 3.7, or 2) add a check when calling `matching_nuclide.append` to not append if an item is already there.